### PR TITLE
Tasks Done

### PR DESCRIPTION
**Pull request by IgorLo**
Remove line if task was accepted earlier.

Mark task as done `[x]` if your resolve it.

- [x] 1 ArrayDequeSimple
- [x] 2 ArrayDequeSimple + Iterator Remove
- [x] 3 ArrayDequeFull
- [x] 4 LinkedDequeSimple
- [x] 5 LinkedDequeSimple + Iterator Remove
- [x] 6 LinkedDequeFull
- [x] 7 ArrayPriorityQueueSimple
- [ ] 8 ArrayPriorityQueueSimple + Iterator Remove
- [x] 9 MergingPeekingIncreasingIterator
- [x] 10 AVLTree
- [x] 11 AVLTreeIterable
- [x] 12 AVLTreeIterable + Iterator Remove
- [ ] 13 RedBlackTree (add)
- [ ] 14 RedBlackTree (add + remove)
- [ ] 15 OpenHashTable + Student
- [ ] 16 OpenHashTable + Student + Iterator Remove

### CODE REVIEW

**Comment by mikhail-nechaev** [comment]
> This branch has conflicts that must be resolved

не смерджено с мастером / бранчем IgorLo

`git pull base IgorLo`

**Comment by mikhail-nechaev** [comment]
без проверки пока есть конфликты.

**Comment by IgorLo** [comment]
Исправил конфликт с мастером

File: src/ru/mail/polis/collections/list/todo/ArrayDequeSimple.java [hunk 1]
```diff
@@ -2,17 +2,106 @@
...
 /**
- * Resizable cyclic array implementation of the {@link IDeque} interface.
- * - no capacity restrictions
+ * Resizable cyclic values implementation of the {@link IDeque} interface.
+ * - no length restrictions
  * - grow as necessary to support
  *
  * @param <E> the type of elements held in this deque
  */
 public class ArrayDequeSimple<E> implements IDeque<E> {
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/list/todo/ArrayDequeSimple.java]: Списано у LoginovArkadiy

**Comment by mikhail-nechaev** [comment]
списано. не принимается
- [ ] 1 ArrayDequeSimple
- [ ] 2 ArrayDequeSimple + Iterator Remove
- [ ] 3 ArrayDequeFull

File: src/ru/mail/polis/collections/iterator/todo/MergingPeekingIncreasingIterator.java [hunk 2]
```diff
@@ -10,46 +15,61 @@
...
      * Complexity = O(n)
      *
      * @param iterators the iterators whose are to be placed into this merging peeking increasing iterator
      */
     public MergingPeekingIncreasingIterator(IntegerIncreasingSequencePeekingIterator... iterators) {
-        //todo: do some stuff with iterators
+        //It takes more than O(n) to build PriorityQueue (N * logN)
+        this.iterators = new ArrayPriorityQueueSimple<>();
+        for (IntegerIncreasingSequencePeekingIterator iterator : iterators) {
+            this.iterators.add(iterator);
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/iterator/todo/MergingPeekingIncreasingIterator.java]: неправильная асимптотика

**Comment by mikhail-nechaev** [comment]
списано. не принимается. a91def442a105444623322ff25c6cb1c37d3f26a
- [ ] 4 LinkedDequeSimple
- [ ] 5 LinkedDequeSimple + Iterator Remove
- [ ] 6 LinkedDequeFull

**Comment by mikhail-nechaev** [comment]
списано. не принимается. 7a3252590e026531c8cf23679379a84a3eb85cfa  и далее
- [ ] 10 AVLTree
- [ ] 11 AVLTreeIterable
- [ ] 12 AVLTreeIterable + Iterator Remove

File: src/ru/mail/polis/collections/list/todo/ArrayPriorityQueueSimple.java [hunk 3]
```diff
@@ -43,69 +47,140 @@ public ArrayPriorityQueueSimple(Collection<E> collection) {
...
      * @throws NullPointerException if the specified collection or comparator is null
      */
     public ArrayPriorityQueueSimple(Collection<E> collection, Comparator<E> comparator) {
+        if (collection == null || comparator == null) {
+            throw new NullPointerException();
+        }
         this.comparator = Objects.requireNonNull(comparator, "comparator");
+        Iterator<E> iterator = collection.iterator();
+        while (iterator.hasNext()) {
+            add(iterator.next());
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/list/todo/ArrayPriorityQueueSimple.java]: не `Complexity = O(n)`

File: test/ru/mail/polis/collections/TestHelper.java [hunk 4]
```diff
@@ -2,23 +2,24 @@
...
+import java.util.stream.Collectors;
 
 /*
  * Created by Nechaev Mikhail
  * Since 07/12/2018.
  */
 public class TestHelper {
 
-    public static final Collection<Object[]> COMPARATORS = Set.of(new Object[][]{
+    public static final Object[][] kek = new Object[][]{
```
Comment (mikhail-nechaev) [inline test/ru/mail/polis/collections/TestHelper.java]: м

**Pull request merged**